core/config/project_settings.cpp

```diff
@@ -647,15 +647,14 @@ Error ProjectSettings::_load_settings_binary(const String &p_path) {
 	}
 
 	f->close();
```

Comment:
Could be done in a follow-up PR if we want to keep things simple here, but `close()` happens in the destructor for `FileAccess` and `DirAccess` so we could remove explicit calls to it, unless it's in places where we explictly want to release the file handle before the RefCounted goes out of scope (e.g. to reopen a file with Write instead of Read).

Reply:
Every single `FileAccess` seems to call `close` in the destructor already, so probably all `f->close();` are redundant.

Reply:
Yeah, if we remove them we should likely make it clear in the documentation that `FileAccess` implementations **should** close in the destructor, that should be part of the contract (not sure if we can validate it somehow).